### PR TITLE
test(Button): add accessibility unit tests and docs

### LINKED ISSUE #7363: Update/overhaul Button accessibility testing
Definition of done: 
- [ ] convert existing tests to React Testing Library
- [ ] add Accessibility Checker test
- [ ] determine and implement missing WCAG 2.1 tests
- [ ] determine and document manual testing practices

### PR DESCRIPTION
**Pull request by dakahn**
Closes #7363 

- adds automated testing with Axe and Accessibility Checker
- adds WCAG accessibility unit tests
- adds manual testing documentation

#### Testing / Reviewing

run `yarn test Button-test.js` and see that there are no errors

### CODE REVIEW

File: packages/react/src/components/Button/button-avt.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## Contrast
+
+- [ ] the Button text has a contrast of 4.5:1 minimum against the background
+      color
+- [ ] the Button background has a contrast of 4.5:1 minimum against the page
+      background
+
+## Screen reader
+
+Each screen reader should be tested when paired with it's preferred browser.
```
Comment (emyarod) [inline packages/react/src/components/Button/button-avt.md]:
```suggestion
Each screen reader should be tested when paired with its preferred browser.
```

**Review by emyarod — COMMENTED** [review]
will the markdown documents need to be updated manually when components are modified?

**Comment by dakahn** [comment]
@emyarod they shouldn't honestly change that much over time, but yeah if they do it's a manual update. Scenarios I can imagine where they'd need an update are if we add a screen reader/browser combo to our testing or there was some weird a11y case that broke and couldn't realistically be covered by unit testing.

**Review by andreancardona — APPROVED** [review]
@dakahn all tests passed - looks good :) 🦖

**Review by joshblack — APPROVED** [review]

**Pull request merged**